app/views/membership_applications/_reason_waiting.html.haml

```diff
@@ -2,32 +2,30 @@
   Do not make user press a "save" or "submit" button for these changes. (use AJAX)
 
 
-= form_for @membership_application, html: { id: 'member-app-waiting-reasons', multipart: true}, url: membership_application_path do |f|
-  .row
-    .container
-      .reason-waiting-information
-        = f.label( :member_app_waiting_reasons, t('membership_applications.need_info.reason_title'))
-        = f.collection_select(:member_app_waiting_reasons_id,
-                              AdminOnly::MemberAppWaitingReason.all,
-                              :id,
-                              reason_name_method,
-                              { include_blank: t('membership_applications.need_info.select_a_reason') },
-                              { class: 'reason-waiting-list',
-                                onchange: 'changed_reason()' },
-                              )
-
-  .row
-    .container
-      #other-text-field
-        = f.label( :reason_waiting_custom_text, t('membership_applications.need_info.other_reason_label') )
-        = f.text_field :custom_reason_text, onchange: 'changed_custom_text()'
+.row
+  .container
+    .reason-waiting-information
+      = label_tag :member_app_waiting_reasons, t('membership_applications.need_info.reason_title')
+      - collection = AdminOnly::MemberAppWaitingReason.all.to_a
+      - collection << AdminOnly::MemberAppWaitingReason.new(id: -1, "name_#{I18n.locale.to_s}": "#{@other_waiting_reason_text}")
+      = select_tag(:member_app_waiting_reasons,
+                   options_from_collection_for_select(collection, :id, reason_name_method,
+                                      @membership_application.member_app_waiting_reasons_id),
+                   { include_blank: t('membership_applications.need_info.select_a_reason'),
+                     class: 'reason-waiting-list' })
+
+.row
+  .container
+    #other-text-field
+      = label_tag :custom_reason_text, t('membership_applications.need_info.other_reason_label')
+      = text_field_tag :custom_reason_text, @membership_application.custom_reason_text
```

Comment:
I removed the form as it was not needed and also caused bugs.  (in your feature branch, enter a custom reason and hit "enter" key - you will see an error alert - this is because _after_ the AJAX stuff happens, the form is also submitted to the controller action (*not* what we want).

I added the "custom reason" item to the select_tag collection.  I hacked it up here but it should probably be returned from a helper method.

I also added the currently-selected option to the options collection.

I refactored the "onchange" stuff to the DOM-ready function.  

I also simplified the ID for the select list and the text_field.  (this make for easier read of the code, IMO)